plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java

```diff
@@ -299,11 +377,27 @@ public void primeEditorValue(@NotNull DBRProgressMonitor monitor, @NotNull Style
                     }
                     applyEditorStyle();
                 }
+            }
+        });
+    }
+
+    private void showRestrictedContent(@NotNull DBDContent value) throws DBCException, IOException {
+        int maxContentSize = DBWorkbench.getPlatform().getPreferenceStore().getInt(ResultSetPreferences.RS_EDIT_MAX_TEXT_SIZE);
+        DBDContentStorage contents = value.getContents(new VoidProgressMonitor());
+        try (final InputStream stream = contents.getContentStream()) {
+            ByteArrayOutputStream buffer = new ByteArrayOutputStream((int) contents.getContentLength());
+            IOUtils.copyStream(stream, buffer);
+            byte[] displaingContentBytes = Arrays.copyOfRange(buffer.toByteArray(), 0, maxContentSize * 1000);
```

Comment:
Can we read `maxContentSize` instead of the entire blob first and then slice it? It might be memory-expensive and won't fit in a 4GB boundary for big BLOBs.

Reply:
Yeap sure, let rework to read require content lenght.